### PR TITLE
Madison/new form queries

### PR DESCRIPTION
**Pull request by madisonmay**

### CODE REVIEW

File: indico/queries/forms.py [hunk 1]
```diff
@@ -58,3 +56,58 @@ def requests(self):
...
+    mutation listPrebuiltFormsMutation {
+        listPrebuiltForms {
+            forms {
+                id
+                name
+            }
+        }
+    }
+    """
+    method = HTTPMethod.GET
```
Comment (nkassis) [inline indico/queries/forms.py:73]: This should be post all graphql requests are POST

File: indico/queries/forms.py [hunk 2]
```diff
@@ -58,3 +56,58 @@ def requests(self):
         else:
             yield UploadDocument(files=self.files)
         yield _FormPreprocessing(files=self.previous)
+
+
+class ListPrebuiltForms(GraphQLRequest):
+
+    query = """
+    mutation listPrebuiltFormsMutation {
```
Comment (nkassis) [inline indico/queries/forms.py:64]: by the name this should be a query not a mutation.

File: indico/queries/forms.py [hunk 3]
```diff
@@ -58,3 +56,58 @@ def requests(self):
...
+        super().__init__(query=self.query)
+
+    def process_response(self, response):
+        return super().process_response(response)["listPrebuiltForms"]["forms"]
+
+
+class GetPrebuiltForm(GraphQLRequest):
+
+    query = """
+    mutation($form_id: Int!) {
```
Comment (nkassis) [inline indico/queries/forms.py:85]: Same as above I think there some confusion on the way graphql works if these are built as mutations.

File: indico/queries/forms.py [hunk 4]
```diff
@@ -58,3 +56,58 @@ def requests(self):
...
+                    bottom
+                    left
+                    right 
+                    pageNum
+                }
+            }
+        }
+    }
+    """
+    method = HTTPMethod.GET
```
Comment (nkassis) [inline indico/queries/forms.py:107]: remove

File: indico/queries/forms.py [hunk 5]
```diff
@@ -1,10 +1,11 @@
 from typing import List
 from pathlib import Path
 
-from indico.client.request import RequestChain, GraphQLRequest
+from indico.client.request import RequestChain, GraphQLRequest, HTTPMethod
```
Comment (jkgenser) [inline indico/queries/forms.py:4]: unused import i believe

**Comment by jkgenser** [comment]
We should add python types to convert the results of the new queries back into. And also add unit tests. See for example: https://github.com/IndicoDataSolutions/indico-client-python/blob/master/indico/queries/datasets.py#L53

**Pull request closed without merging**